src/components/modal.tsx

```diff
@@ -1,12 +1,77 @@
-import Modal from 'react-modal'
+import { faTimes } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import React from 'react'
+import BaseModal from 'react-modal'
+import styled from 'styled-components'
+
+import { StyledH2 } from './tags/styled-h2'
+import { useInstanceData } from '@/contexts/instance-context'
 
 try {
-  Modal.defaultStyles.overlay!.zIndex = 101
+  BaseModal.defaultStyles.overlay!.zIndex = 101
 } catch (e) {
   console.log(e)
   //
 }
 
-Modal.setAppElement('#__next')
+BaseModal.setAppElement('#__next')
+
+export function ModalWithCloseButton({
```

Comment:
Since the component is the main export the file should be called modal-with-close-button.tsx to follow serlo naming convention.